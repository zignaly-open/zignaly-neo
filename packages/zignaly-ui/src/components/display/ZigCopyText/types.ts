import { ZigInputProps } from "../../inputs/ZigInput/types";

export type ZigCopyTextProps = Omit<
  ZigInputProps,
  "onChange" | "href" | "readonly" | "disabled" | "labelAction" | "value"
> & {
  value: string;
  copyElementId?: string;
  onCopied?: () => void;
};
