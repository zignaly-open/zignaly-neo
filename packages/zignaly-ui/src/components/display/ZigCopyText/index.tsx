import React from "react";
import { ZigCopyTextProps } from "./types";
import { styled } from "@mui/material/styles";
import ZigInput, { ZigInputInteractiveAdornmentStyle } from "../../inputs/ZigInput";
import { InputAdornment } from "@mui/material";
import ContentCopyIcon from "@mui/icons-material/ContentCopy";
import copy from "copy-to-clipboard";

const ZigCopyText: React.FC<ZigCopyTextProps> = styled<React.FC<ZigCopyTextProps>>(
  ({ onCopied, copyElementId, InputProps, ...props }) => {
    const doCopy = () => {
      onCopied?.();
      copy(props.value as string);
    };
    return (
      <ZigInput
        wide
        {...props}
        InputProps={{
          ...InputProps,
          endAdornment: (
            <InputAdornment position="end">
              <ContentCopyIcon
                id={copyElementId}
                onClick={doCopy}
                width={40}
                height={40}
                sx={ZigInputInteractiveAdornmentStyle}
              />
            </InputAdornment>
          ),
        }}
        readOnly
      />
    );
  },
)`
  .MuiInput-root {
    &,
    & .MuiInput-input {
      cursor: default;
    }

    &,
    &.Mui-focused,
    &:hover {
      border-color: ${({ theme }) => theme.palette.neutral700} !important;
    }
  }
`;

export default ZigCopyText;
