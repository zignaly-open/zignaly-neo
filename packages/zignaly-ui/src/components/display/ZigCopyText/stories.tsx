import React from "react";
import "@mui/system";
import { ComponentMeta, ComponentStory } from "@storybook/react";
import ZigCopyText from "./";
import "@mui/system";

const result: ComponentMeta<typeof ZigCopyText> = {
  title: "Display/ZigCopyText",
  component: ZigCopyText,
  argTypes: {
    label: {
      type: "string",
    },
    value: {
      type: "string",
    },
  },
};

export default result;

const Template: ComponentStory<typeof ZigCopyText> = (args) => (
  <ZigCopyText onCopied={() => alert("Copied!")} {...args} />
);

export const CopyText: ComponentMeta<typeof ZigCopyText> = Template.bind({});
CopyText.args = {
  label: "Your deposit address",
  value: "0xjhkjfhvjkhdskjvhdfskjvhsdfkjhvjkh",
};

export const CopySensitiveText: ComponentMeta<typeof ZigCopyText> = Template.bind({});
CopySensitiveText.args = {
  label: "Your deposit address",
  sensitive: true,
  value: "0xjhkjfhvjkhdskjvhdfskjvhsdfkjhvjkh",
};
