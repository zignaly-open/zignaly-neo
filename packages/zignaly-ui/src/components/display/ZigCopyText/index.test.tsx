import React from "react";
import { render, fireEvent } from "@testing-library/react";
import ZigCopyText from ".";

describe("components/display/ZigCopyText", () => {
  it("should render without crashing", () => {
    const { container } = render(<ZigCopyText id="test-id" value="test value" />);

    const input = container.querySelector("#test-id") as HTMLElement;
    expect(input).toBeVisible();
    expect(input).toMatchSnapshot();
  });

  it("should have a copy button", () => {
    const { container } = render(
      <ZigCopyText id="test-id" value="test value" copyElementId="copy-button-id" />,
    );

    const copyButton = container.querySelector("#copy-button-id") as HTMLElement;
    expect(copyButton).toBeInTheDocument();
  });

  it("should call onCopied when copy button is clicked", () => {
    const handleCopied = jest.fn();

    const { container } = render(
      <ZigCopyText
        id="test-id"
        value="test value"
        copyElementId="copy-button-id"
        onCopied={handleCopied}
      />,
    );

    const copyButton = container.querySelector("#copy-button-id") as HTMLElement;

    fireEvent.click(copyButton);

    expect(handleCopied).toHaveBeenCalled();
  });
});
