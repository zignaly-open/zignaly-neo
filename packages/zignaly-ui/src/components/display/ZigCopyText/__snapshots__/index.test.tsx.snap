// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/display/ZigCopyText should render without crashing 1`] = `
<input
  aria-invalid="false"
  class="MuiInputBase-input MuiInput-input MuiInputBase-inputAdornedEnd css-1x51dt5-MuiInputBase-input-MuiInput-input"
  data-testid="test-id"
  id="test-id"
  type="text"
  value="test value"
/>
`;
